File: backend/src/main/java/com/quickbite/quejas/service/SucursalService.java
package com.quickbite.quejas.service;

import com.quickbite.quejas.dto.sucursal.SucursalRequest;
import com.quickbite.quejas.dto.sucursal.SucursalResponse;

import java.util.List;

/** CU05 - Administrar Sucursales. */
public interface SucursalService {
    List<SucursalResponse> listar();
    SucursalResponse crear(SucursalRequest request);
    SucursalResponse actualizar(Long id, SucursalRequest request);
    SucursalResponse cambiarEstado(Long id);
}

File: frontend/src/app/core/services/sucursal.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { SucursalResponse, SucursalRequest } from '../models/sucursal.model';

/** CU05 - Administrar Sucursales. */
@Injectable({ providedIn: 'root' })
export class SucursalService {
  private readonly baseUrl = `${environment.apiUrl}/sucursales`;

  constructor(private http: HttpClient) {}

  listar(): Observable<SucursalResponse[]> {
    return this.http.get<SucursalResponse[]>(this.baseUrl);
  }

  crear(request: SucursalRequest): Observable<SucursalResponse> {
    return this.http.post<SucursalResponse>(this.baseUrl, request);
  }

  actualizar(id: number, request: SucursalRequest): Observable<SucursalResponse> {
    return this.http.put<SucursalResponse>(`${this.baseUrl}/${id}`, request);
  }

  cambiarEstado(id: number): Observable<SucursalResponse> {
    return this.http.patch<SucursalResponse>(`${this.baseUrl}/${id}/estado`, {});
  }
}
